File: back/src/main/java/com/ayd2/congress/services/Organization/OrganizationService.java
package com.ayd2.congress.services.Organization;

import java.util.List;

import com.ayd2.congress.dtos.Organization.NewOrganizationRequest;
import com.ayd2.congress.dtos.Organization.OrganizationResponse;
import com.ayd2.congress.dtos.Organization.OrganizationUpdate;
import com.ayd2.congress.exceptions.DuplicatedEntityException;
import com.ayd2.congress.exceptions.NotFoundException;
import com.ayd2.congress.models.Organization.OrganizationEntity;

public interface OrganizationService {
    OrganizationResponse create(NewOrganizationRequest request) throws DuplicatedEntityException;
    OrganizationEntity getById(Long id) throws NotFoundException;
    List<OrganizationResponse> getAll();
    OrganizationResponse update(OrganizationUpdate request,Long id) throws NotFoundException, DuplicatedEntityException;
}

File: front/src/app/Services/Organization/organization.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environemnt } from '../../../environment/Environment';
import { NewOrganizationRequest, Organization, OrganizationUpdate } from '../../interfaces/Organization';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class OrganizationService {
  baseUrl = environemnt.baseUrl + '/organizations'
  constructor(private http:HttpClient) { }
  create(request: NewOrganizationRequest): Observable<Organization> {
    return this.http.post<Organization>(this.baseUrl, request);
  }

  getAll(): Observable<Organization[]> {
    return this.http.get<Organization[]>(this.baseUrl);
  }

  getById(id: number): Observable<Organization> {
    return this.http.get<Organization>(`${this.baseUrl}/${id}`);
  }

  update(id: number, request: OrganizationUpdate): Observable<Organization> {
    return this.http.put<Organization>(`${this.baseUrl}/${id}`, request);
  }
}
